packages/opentelemetry-plugin-http/src/http.ts

```diff
@@ -0,0 +1,414 @@
+    ) {
+      shimmer.wrap(
+        this._moduleExports.Server.prototype,
+        'emit',
+        this._getPatchIncomingRequestFunction()
+      );
+    } else {
+      this._logger.error(
+        'Could not apply patch to %s.emit. Interface is not as expected.',
+        this.moduleName
+      );
+    }
+
+    return this._moduleExports;
+  }
+
+  /** Unpatches all HTTP patched function. */
+  protected unpatch(): void {
+    shimmer.unwrap(this._moduleExports, 'request');
+    if (semver.satisfies(this.version, '>=8.0.0')) {
+      shimmer.unwrap(this._moduleExports, 'get');
+    }
+    if (
+      this._moduleExports &&
+      this._moduleExports.Server &&
+      this._moduleExports.Server.prototype
+    ) {
+      shimmer.unwrap(this._moduleExports.Server.prototype, 'emit');
+    }
+  }
+
+  /**
+   * Creates spans for incoming requests, restoring spans' context if applied.
+   */
+  protected _getPatchIncomingRequestFunction() {
+    return (original: (event: string) => boolean) => {
+      return this.incomingRequestFunction(original);
+    };
+  }
+
+  /**
+   * Creates spans for outgoing requests, sending spans' context for distributed
+   * tracing.
+   */
+  protected _getPatchOutgoingRequestFunction() {
+    return (original: Func<ClientRequest>): Func<ClientRequest> => {
+      return this.outgoingRequestFunction(original);
+    };
+  }
+
+  protected _getPatchOutgoingGetFunction() {
+    return (original: Func<ClientRequest>): Func<ClientRequest> => {
+      // Re-implement http.get. This needs to be done (instead of using
+      // getPatchOutgoingRequestFunction to patch it) because we need to
+      // set the trace context header before the returned ClientRequest is
+      // ended. The Node.js docs state that the only differences between
+      // request and get are that (1) get defaults to the HTTP GET method and
+      // (2) the returned request object is ended immediately. The former is
+      // already true (at least in supported Node versions up to v10), so we
+      // simply follow the latter. Ref:
+      // https://nodejs.org/dist/latest/docs/api/http.html#http_http_get_options_callback
+      // https://github.com/googleapis/cloud-trace-nodejs/blob/master/src/plugins/plugin-http.ts#L198
+      return function outgoingGetRequest(
+        options: string | RequestOptions | URL,
+        callback?: HttpCallback
+      ) {
+        const req = request(options, callback);
+        req.end();
+        return req;
+      };
+    };
+  }
+
+  /**
+   * Injects span's context to header for distributed tracing and finishes the
+   * span when the response is finished.
+   * @param request The original request object.
+   * @param options The arguments to the original function.
+   */
+  private getMakeRequestTraceFunction(
+    request: ClientRequest,
+    options: RequestOptions
+  ): Func<ClientRequest> {
+    return (span: Span): ClientRequest => {
+      this._logger.debug('makeRequestTrace');
+
+      if (!span) {
```

Comment:
IMO this is superfluous falsiness check, `span` can't be `null` or `undefined` here. WDYT?

Reply:
I could be null. Anyway, for now, span is not passed to the function so it was always null... I refactored, so now you're right. But I asked a question on gitter https://gitter.im/open-telemetry/opentelemetry-node